src/units/abilities.cpp

```diff
@@ -199,11 +199,21 @@ int find_direction(const map_location& loc, const map_location& from_loc, std::s
 }
 
 /**
- * This function return true if locations checked are the same, or if units have same id.
+ * These functions return true if @a u not incapacitated, if @a u underlying_id() not equals to *this underlying_id() and other critera.
```

Comment:
I refer to my [previous comment](https://github.com/wesnoth/wesnoth/pull/10426#discussion_r2257689727), that it doesn't feel right putting these different concepts together in a function.

Neither the documentation nor the name hint at why a unit would be "valid" when maintaining this file, so reading it without the benefit of seeing that it's part of a PR about affect_adjacent abilities.